### PR TITLE
Added options to disable additional ore generation in Mesa and Extreme Hills

### PR DESCRIPTION
**Pull request by MelonCode**
I had task to generate a world that would not contain any ores.
For this purpose I used a customized generator type with specific preset. However, some biomes, such as extreme hills and mesa, still contained ores.
This happened because generation of these blocks were hard coded into world generation and had no respect to worldgen preset settings

> In mesa biomes, an additional 20 veins generate between layers 32 and 79 (inclusive)
> Extreme hills are the only biomes where emerald ores and silverfish can be found naturally.
Minecraft Wiki

In this patch, I added 3 world options to disable it
```
    generator-settings:
       disable-extreme-hills-emeralds: false
       disable-extreme-hills-monster-eggs: false
       disable-mesa-additional-gold: false
```

### CODE REVIEW

File: Spigot-Server-Patches/0270-Additional-Ore-Generation-Settings-for-Mesa-and-Extr.patch [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+-
+ import com.destroystokyo.paper.antixray.ChunkPacketBlockControllerAntiXray.ChunkEdgeMode;
+ import com.destroystokyo.paper.antixray.ChunkPacketBlockControllerAntiXray.EngineMode;
+ import net.minecraft.server.MinecraftServer;
+@@ -10,6 +7,9 @@ import org.bukkit.Bukkit;
+ import org.bukkit.configuration.file.YamlConfiguration;
+ import org.spigotmc.SpigotWorldConfig;
+ 
++import java.util.Arrays;
++import java.util.List;
```
Comment (zachbr) [inline Spigot-Server-Patches/0270-Additional-Ore-Generation-Settings-for-Mesa-and-Extr.patch]: I assume your IDE did this.

File: Spigot-Server-Patches/0270-Additional-Ore-Generation-Settings-for-Mesa-and-Extr.patch [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+     public boolean optimizeExplosions;
+diff --git a/src/main/java/net/minecraft/server/BiomeBigHills.java b/src/main/java/net/minecraft/server/BiomeBigHills.java
+index 9c39bf7..5bab977 100644
+--- a/src/main/java/net/minecraft/server/BiomeBigHills.java
++++ b/src/main/java/net/minecraft/server/BiomeBigHills.java
+@@ -32,23 +32,25 @@ public class BiomeBigHills extends BiomeBase {
+         int k;
+         int l;
+ 
+-        for (j = 0; j < i; ++j) {
```
Comment (zachbr) [inline Spigot-Server-Patches/0270-Additional-Ore-Generation-Settings-for-Mesa-and-Extr.patch]: Wrap this but _don't_ indent.
Minimal diff takes precedence.

File: Spigot-Server-Patches/0270-Additional-Ore-Generation-Settings-for-Mesa-and-Extr.patch [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
++            }
+ 
+     }
+ 
+@@ -70,6 +72,7 @@ public class BiomeBigHills extends BiomeBase {
+ 
+         NORMAL, EXTRA_TREES, MUTATED;
+ 
+-        private Type() {}
++        private Type() {
```
Comment (zachbr) [inline Spigot-Server-Patches/0270-Additional-Ore-Generation-Settings-for-Mesa-and-Extr.patch]: same as above

**Comment by zachbr** [comment]
Hey so thanks for the PR.

There were some minor formatting quibbles that I've noted but handled myself. I don't feel they were anything super big or annoying to make you handle. One of those was merging the patch back into the original. We usually don't like to keep changes to an original patch separate from that original patch file.

I've also rebased this against latest so it can be merged.

**Review by zachbr — APPROVED** [review]

**Comment by MelonCode** [comment]
Thank you, was glad to contribute. I'll take note of this next time,

**Pull request merged**